ML/MM with GROMACS and Metatomic

PicoCentauri opened the pull request:
This is in principle complete but we have to make the GROMACS interface available.

Happy to improve if there are already comments.

PicoCentauri committed:
ML/MM with GROMACS and Metatomic

pfebrer commented on src/generate-gallery.py:
```diff
@@ -42,7 +42,7 @@ def __init__(self, example):
             "gallery_dirs": gallery_dir,
             "write_computation_times": False,
-            "copyfile_regex": r".*\.(sh|xyz|cp2k|yml|yaml|jpg|jpeg|png|zip)",
+            "copyfile_regex": r".*\.(cp2k|jpg|jpeg|mdp|sh|yaml|yml|zip)",
```
Why not `xyz` and `png`?

PicoCentauri replied:
ahh yes, sorry. I was cleaning to hard xD

PicoCentauri committed:
Merge branch 'main' into mts-gromacs

HaoZeke committed:
feat(ml-mm): enable metatomic ML/MM with gromacs-metatomic conda package

- Switch from plain gromacs to gromacs-metatomic (metatensor channel)
- Uncomment metatomic MDP section (active=yes, device=cpu)
- Uncomment model export step (PET-MAD v1.0.2 from HuggingFace)
- Add metatomic-torch dependency for model export

PicoCentauri committed:
Merge branch 'main' into mts-gromacs

ceriottm committed:
Merge branch 'main' into mts-gromacs

HaoZeke committed:
feat(ml-mm): use PET-MAD XS v1.5.0 and add Ramachandran analysis

- Switch to PET-MAD XS v1.5.0 from lab-cosmo/upet (smaller, faster)
- Use the mtt export style matching other cookbook recipes (eon-pet-neb)
- Add Ramachandran phi/psi dihedral analysis for alanine dipeptide
- Add matplotlib dependency and side-by-side RMSD + Ramachandran plot
- Better structured narrative with section headers

Co-authored-by: Rohit Goswami <rohit.goswami@epfl.ch>

PicoCentauri committed:
proofread

pfebrer commented on examples/ml-mm/ml-mm.py:
```diff
@@ -0,0 +1,220 @@
+:Authors: Philip Loche `@PicoCentauri <https://github.com/PicoCentauri>`_,
+          Rohit Goswami `@HaoZeke <https://github.com/haozeke/>`_
+
+In this tutorial we simulate and analyse a alanine dipeptide in water using a machine
```
```suggestion
In this tutorial we simulate and analyse an alanine dipeptide in water using a machine
```

pfebrer commented on examples/ml-mm/ml-mm.py:
```diff
@@ -0,0 +1,220 @@
+#
+# Before running the simulation, we need to export the ML model into the TorchScript
+# format that GROMACS can load. We download the PET-MAD XS checkpoint from HuggingFace
+# and export it using ``Metatrain``.
```
```suggestion
# and export it using ``metatrain``.
```

pfebrer reviewed (changes requested):
Two super important changes

pfebrer committed:
Fixed my own suggestions

Because my suggestions are always good
